.circleci/tests/test_despike.py

```diff
@@ -17,3 +25,93 @@ def test_data_availability(data_dir, working_dir, output_dir):
+    boldfile = data_dir + "/withoutfreesurfer/sub-01/func/" \
+        "sub-01_task-mixedgamblestask_run-1_space-MNI152NLin2009cAsym_desc-preproc_bold.nii.gz"
+    maskfile = data_dir + "/withoutfreesurfer/sub-01/func/" \
+        "sub-01_task-mixedgamblestask_run-1_space-MNI152NLin2009cAsym_desc-brain_mask.nii.gz"
+    # Create some spikes in the second voxel
+    file_data = read_ndata(boldfile, maskfile)
+    voxel_data = file_data[2, :]  # a voxel across time
+    voxel_data_mean = np.mean(voxel_data)
+    voxel_data_std = np.std(voxel_data)
+    voxel_data[2] = voxel_data_mean + 10 * voxel_data_std
+    voxel_data[3] = voxel_data_mean - 10 * voxel_data_std
+    # What's the min and max, i.e: the amplitude of the spikes created?
+    spiked_min = min(voxel_data)
+    spiked_max = max(voxel_data)
+    # Let's write this temp file out for despiking
+    file_data[2, :] = voxel_data
+    tempdir = tempfile.mkdtemp()
+    filename = tempdir + "/spikedfile.nii.gz"
+    write_ndata(data_matrix=file_data, mask=maskfile, template=boldfile, TR=0.8, filename=filename)
+    spikedfile = filename
+    # Let's despike the image and write it out to a temp file
+    tempdir = tempfile.mkdtemp()
+    os.system("3dDespike -NEW -prefix " + tempdir + "/3dDespike.nii.gz " + spikedfile)
+    despiked_file = tempdir + "/3dDespike.nii.gz"
+    file_data = read_ndata(despiked_file, maskfile)
+    voxel_data = file_data[2, :]
+    # What's the min and max of the despiked file?
+    despiked_min = min(voxel_data)
+    despiked_max = max(voxel_data)
+    # Have the spikes been reduced?
+    # i.e: has the minimum increased after despiking, and the maximum decreased?
+    # Additionally, have the affines changed?
+    assert spiked_min < despiked_min
+    assert spiked_max > despiked_max
+    assert np.array_equal(nb.load(despiked_file).affine, nb.load(boldfile).affine)
+
+    return
+
+
+def test_cifti_despike(data_dir):
+    """Test Cifti despiking."""
+    boldfile = data_dir + "/fmriprep/sub-colornest001/ses-1/func/"\
+        "sub-colornest001_ses-1_task-rest_run-1_space-fsLR_den-91k_bold.dtseries.nii"
+    # Let's add some noise
+    file_data = read_ndata(boldfile)
+    voxel_data = file_data[2, :]  # a voxel across time
+    voxel_data_mean = np.mean(voxel_data)
+    voxel_data_std = np.std(voxel_data)
+    voxel_data[2] = voxel_data_mean + 10 * voxel_data_std
+    voxel_data[3] = voxel_data_mean - 10 * voxel_data_std
+    # What's the maximum and minimum values of the data?
+    spiked_max = max(voxel_data)
+    spiked_min = min(voxel_data)
+    # Let's write this out
+    file_data[2, :] = voxel_data
+    tempdir = tempfile.mkdtemp()
+    filename = tempdir + "/test.nii"
+    write_ndata(data_matrix=file_data, template=boldfile, TR=0.8, filename=filename)
+    # Let's despike the data
+    # Run the node the same way it's run in XCP
+    in_file = filename
+    TR = _get_tr(nb.load(filename))
+    tempdir = tempfile.mkdtemp()
```

Comment:
I generally prefer to use the pytest `tmp_path` fixture instead of trying to manage the temporary paths within the test.

Reply:
will have a look!